ddtrace/opentelemetry/span_test.go

```diff
@@ -22,38 +23,236 @@ import (
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/httpmem"
 )
 
-func TestSpanMethods(t *testing.T) {
-	testData := struct {
-		env, srv, oldOp, newOp string
-		tags                   [][]string // key - tags[0][0], value - tags[0][1]
-	}{
-		env:   "test_env",
-		srv:   "test_srv",
-		oldOp: "old_op",
-		newOp: "new_op",
-		tags:  [][]string{{"tag_1", "tag_1_val"}, {"opt_tag_1", "opt_tag_1_val"}}}
-	done := make(chan struct{})
-	var payload string
+// helper method to wait until timeout for test agent to receive traces
+func waitForTestAgent(done chan struct{}, timeOut time.Duration, t *testing.T) {
+	select {
+	case <-time.After(timeOut):
+		t.FailNow()
+	case <-done:
+		break
+	}
+}
+
+// initializes a test trace provider and agent server
+func getTestTracerProvider(payload *string, done chan struct{},
+	env string, service string, t *testing.T) (*TracerProvider, *http.Server) {
 	s, c := httpmem.ServerAndClient(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		switch r.URL.Path {
 		case "/v0.4/traces":
 			buf, err := io.ReadAll(r.Body)
 			if err != nil {
 				t.Fail()
 			}
-			payload = fmt.Sprintf("%s", buf)
-			done <- struct{}{}
+			*payload = fmt.Sprintf("%s", buf)
+			if strings.Contains(*payload, "name") {
+				// tracer.Start() also sends a request to /v0.4/traces
+				// but does not call waitForTestAgent causing this to block
+				done <- struct{}{}
+			}
```

Comment:
The suggestion functionality doesn't let me display this better, but this should be the body of that switch case entirely.
```suggestion
			// tracer.Start() also sends a request to /v0.4/traces with empty payload
			// but does not call waitForTestAgent causing this to block
			if h := r.Header.Get("X-Datadog-Trace-Count"); h == "0" {
				return
			}
			buf, err := io.ReadAll(r.Body)
			if err != nil {
				t.Fail()
			}
			*payload = fmt.Sprintf("%s", buf)
			done <- struct{}{}
			}
```